ironic_prometheus_exporter/parsers/ipmi.py

```diff
@@ -71,17 +79,28 @@ def extract_labels(entries, payload, node_name):
     LOG.info('extract_labels function called with: entries=%s | payload=%s | \
              node_name=%s' % (str(entries), str(payload), node_name))
     if len(entries) == 1:
+        status = payload[entries[0]].get('Status')
         labels = {'node_name': node_name,
-                  'entity_id': payload[entries[0]]['Entity ID']}
+                  'node_uuid': node_uuid,
+                  'instance_uuid': instance_uuid,
```

Comment:
I don't think instance_uuid is going to be useful, and since each label adds to the storage and processing costs in prometheus we should probably drop it.

Reply:
@dhellmann ack, going to drop it

Reply:
If there's a 1:1 mapping between the labels, it isn't a big problem for Prometheus: it will make the index a little bigger but it won't increase the metrics cardinality. Looking at the test data though, it seems that `node_uuid` and `instance_uuid` have the same values so there's no need to expose both indeed.

Having said that, when an application has labels that are common for every timeseries, you can expose a constant gauge metric with these labels like:

```
baremetal_machine_info{node_uuid="ac2aa2fd-6e1a-41c8-a114-2084c8705228", node_name="knilab-master-u8"} 1
...
baremetal_front_led_panel{node_uuid="ac2aa2fd-6e1a-41c8-a114-2084c8705228",entity_id="7.1 (System Board)"} 0
```

Then you can use PromQL's vector matches to access the additional labels:

```
# this query will return:
# {instance="localhost:8081",job="app",node_name="knilab-master-u8",node_uuid="ac2aa2fd-6e1a-41c8-a114-2084c8705228"} 0
baremetal_front_led_panel * on(job,instance,node_uuid) group_right baremetal_machine_info
```

HTH

Reply:
This app is collecting data about more than one ironic node, so the UUID values will change and we will have different stats in the same result set for different nodes. So I think we do want to continue to label each value as we report it.